Include/Internal/type.h

```diff
@@ -19,6 +19,11 @@ class DONSUS_TYPE {
     TYPE_VOID
   };
 
+  bool operator<(
+      const DONSUS_TYPE &other) const { // for the set to order its elements
+    return type_un < other.type_un;
+  }
```

Comment:
Could you elaborate on this? What kind of set needs this overloaded operator?

Reply:
So recently I have updated the code for typechecking expressions, the way we do it now is we consider each element in the expression and add its type to a std::set, then since the set will not store the same items if there are more than two items within the set then that means that there are some type problems, e.g if the set contains string_expr and number_expr then we have an incompatible type. The reason why we need that overload is because by default the std::set will need that in order to compare elements and order them properly.